azure-quantum/azure/quantum/job/session.py

```diff
@@ -139,6 +235,37 @@ def open_session(
         job_failure_policy: Union[str, SessionJobFailurePolicy, None] = None,
         **kwargs
     ) -> Session:
+        """Opens a session and associates all future job submissions to that
+           session until the session is closed (which happens automatically
+           after existing a `with` block).
+
+        Example (job 1 to 3 will be associated the session "MySession"):
+            with target.open_session(name="MySession") as session:
+                job1 = target.submit(input_data=input_data, job_name="Job 1")
+                job2 = target.submit(input_data=input_data, job_name="Job 2")
+                job3 = target.submit(input_data=input_data, job_name="Job 3")
+
+        :param details: Session details model, containing the session id,
+                        name, job_failure_policy, provider_id and target.
+                        Either this parameter should be passed containing all
+                        the session detail values, the same values should be
+                        passed as individual parameters.
+        :type details: Optional[SessionDetails]
+
+        :param id: The id of the session. If not passed, one random uuid will used.
+        :type id: Optional[str]
+
+        :param name: The name of the session.
+                    If not passed, the name will be `session-{session-id}`.
+        :type name: Optional[str]
+
+        :param job_failure_policy: The policy that determines when a session would fail,
+                                close and not accept further jobs.
+        :type job_failure_policy: Union[str, SessionJobFailurePolicy, None]
+
+        :return: The session object with updated details after its opening.
+        :rtype: Session
+        """
         if self.latest_session:
             self.latest_session.close()
```

Comment:
The service already handles gracefully the case where close is issued on a session that is already in a terminal state, but it would be  a nice optimization to avoid making the call if the client knows that it is already closed.

Reply:
Done. Handled in the `workspace.close_session`.